drivers/flash/flash_stm32h7x.c

```diff
@@ -36,11 +36,16 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 #define STM32H7_FLASH_TIMEOUT	\
 	(2 * DT_PROP(DT_INST(0, st_stm32_nv_flash), max_erase_time))
 
+#define STM32H7_M4_FLASH_SIZE DT_PROP_OR(DT_INST(0, st_stm32_nv_flash), bank2_flash_size, 0)
 #ifdef CONFIG_CPU_CORTEX_M4
+#if STM32H7_M4_FLASH_SIZE == 0
 #error Flash driver on M4 core is not supported yet
```

Comment:
Shouldn't the error be adapted? As I understand, this PR adds the support but requires to set the size in DT.

Reply:
Pushed the update @GeorgeCGV